packages/schema/src/language-server/validator/datamodel-validator.ts

```diff
@@ -61,8 +62,13 @@ export default class DataModelValidator implements AstValidator<DataModel> {
                 if (idField.type.optional) {
                     accept('error', 'Field with @id attribute must not be optional', { node: idField });
                 }
-                if (idField.type.array || !idField.type.type || !SCALAR_TYPES.includes(idField.type.type)) {
-                    accept('error', 'Field with @id attribute must be of scalar type', { node: idField });
+
+                const isArray = idField.type.array;
+                const isScalar = SCALAR_TYPES.includes(idField.type.type as typeof SCALAR_TYPES[number])
+                const isValidType = isScalar || isEnum(idField.type.reference?.ref?.$type)
```

Comment:
I believe it should be:

```ts
isEnum(idField.type.reference?.ref)
```

I understand it's a bit confusing 😄. The `idField.type.reference?.ref` part is the AST node that the reference is resolved to. The `$type` should actually give the string "EnumDecl". For consistency I'd suggest to call `isEnum` on the AST node.

Reply:
No worries. It's looking great now!